src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/create_index_pattern_wizard.js

```diff
@@ -1,307 +1,232 @@
-import _ from 'lodash';
-import { IndexPatternMissingIndices } from 'ui/errors';
-import 'ui/directives/validate_index_pattern';
-import 'ui/directives/auto_select_if_only_one';
-import 'ui/directives/documentation_href';
-import uiRoutes from 'ui/routes';
-import { uiModules } from 'ui/modules';
-import template from './create_index_pattern_wizard.html';
-import { sendCreateIndexPatternRequest } from './send_create_index_pattern_request';
-import './step_index_pattern';
-import './step_time_field';
-import './matching_indices_list';
-import './create_index_pattern_wizard.less';
-
-uiRoutes
-  .when('/management/kibana/index', {
-    template,
-  });
-
-uiModules.get('apps/management')
-  .controller('managementIndicesCreate', function (
-    $routeParams,
-    $scope,
-    $timeout,
-    config,
-    es,
-    indexPatterns,
-    kbnUrl,
-    Notifier,
-    Promise
-  ) {
-  // This isn't ideal. We want to avoid searching for 20 indices
-  // then filtering out the majority of them because they are sysetm indices.
-  // We'd like to filter system indices out in the query
-  // so if we can accomplish that in the future, this logic can go away
-    const ESTIMATED_NUMBER_OF_SYSTEM_INDICES = 20;
-    const MAX_NUMBER_OF_MATCHING_INDICES = 20;
-    const MAX_SEARCH_SIZE = MAX_NUMBER_OF_MATCHING_INDICES + ESTIMATED_NUMBER_OF_SYSTEM_INDICES;
-    const notify = new Notifier();
-    const disabledDividerOption = {
-      isDisabled: true,
-      display: '───',
-    };
-    const noTimeFieldOption = {
-      display: `I don't want to use the Time Filter`,
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  EuiSpacer,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiTextColor,
+  EuiSwitch,
+  EuiPanel,
+  EuiIcon,
+  EuiLink,
+  EuiButton,
+} from '@elastic/eui';
+
+import { MAX_SEARCH_SIZE } from './constants';
+import { getIndices } from './lib/get_indices';
+
+export class CreateIndexPatternWizard extends Component {
+  static propTypes = {
+    loadingDataDocUrl: PropTypes.string.isRequired,
+    services: PropTypes.shape({
+      es: PropTypes.object.isRequired,
+      indexPatterns: PropTypes.object.isRequired,
+      config: PropTypes.object.isRequired,
+      kbnUrl: PropTypes.object.isRequired,
+      notify: PropTypes.object.isRequired,
+    }).isRequired,
```

Comment:
What's the advantage of nesting these props inside of a `services` object instead of passing them indirectly?